main.tf

```
@@ -38,47 +50,96 @@ resource "aws_efs_access_point" "default" {
 
   file_system_id = join("", aws_efs_file_system.default.*.id)
 
-  posix_user {
-    gid = var.access_points[each.key]["posix_user"]["gid"]
-    uid = var.access_points[each.key]["posix_user"]["uid"]
-    # Just returning null in the lookup function gives type errors and is not omitting the parameter, this work around ensures null is returned.
-    secondary_gids = lookup(lookup(var.access_points[each.key], "posix_user", {}), "secondary_gids", null) == null ? null : null
+  dynamic "posix_user" {
+    for_each = local.secondary_gids[each.key] != null ? ["true"] : []
+
+    content {
+      gid            = var.access_points[each.key]["posix_user"]["gid"]
+      uid            = var.access_points[each.key]["posix_user"]["uid"]
+      secondary_gids = local.secondary_gids[each.key] != null ? split(",", local.secondary_gids[each.key]) : null
+    }
   }
 
   root_directory {
     path = "/${each.key}"
-    creation_info {
-      owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
-      owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
-      permissions = var.access_points[each.key]["creation_info"]["permissions"]
+
+    dynamic "creation_info" {
+      for_each = try(lookup(var.access_points[each.key]["creation_info"]["gid"], ""), "") != "" ? ["true"] : []
+
+      content {
+        owner_gid   = var.access_points[each.key]["creation_info"]["gid"]
+        owner_uid   = var.access_points[each.key]["creation_info"]["uid"]
+        permissions = var.access_points[each.key]["creation_info"]["permissions"]
+      }
     }
   }
 
   tags = module.this.tags
 }
 
+module "security_group_label" {
+  source  = "cloudposse/label/null"
+  version = "0.25.0"
+
+  # For backwards compatibility with v0.30.1
+  attributes = concat(module.this.attributes, [var.security_group_suffix])
+
+  context = module.this.context
+}
+
 module "security_group" {
   source  = "cloudposse/security-group/aws"
-  version = "0.3.1"
+  version = "0.4.2"
 
-  use_name_prefix = var.security_group_use_name_prefix
-  rules           = var.security_group_rules
-  vpc_id          = var.vpc_id
-  description     = var.security_group_description
+  enabled                       = local.security_group_enabled
+  security_group_name           = var.security_group_name
+  create_before_destroy         = var.security_group_create_before_destroy
+  security_group_create_timeout = var.security_group_create_timeout
+  security_group_delete_timeout = var.security_group_delete_timeout
 
-  enabled = local.security_group_enabled
-  context = module.this.context
+  security_group_description = var.security_group_description
+  allow_all_egress           = true
+  rules                      = var.additional_security_group_rules
+  rule_matrix = [
+    {
+      source_security_group_ids = local.allowed_security_group_ids
+      cidr_blocks               = var.allowed_cidr_blocks
+      rules = [
+        {
+          key         = null
```

`key = null` is bad style. Omit `key` entirely if you cannot set it to something unique. Here, however, since there is no loop, you can set it to whatever you want.
```suggestion
          key         = "in"
```